### PR TITLE
Login API

### LINKED ISSUE #5: Login form & API (POST /login)
In order to authenticate users and allow them to vote, we need to let them login!

## This Issue
- Puts a login form on the root page (`/`) 
- Creates a `POST /login` API, which will:
  - Pretend to check a DB for the user email
  - Redirect to `/ballot` if successful and the user still has votes
  - Redirect `/thank-you` if successful and the user no longer has any votes
  - Redirects to `/` login page if unsuccessful

### PR DESCRIPTION
**Pull request by zackfern**
This PR:
- Puts a login form on the root page (`/`) 
- Creates a `POST /login` API, which will:
  - Check the DB for the user email
  - Redirect to `/ballot` if successful and the user still has votes
  - Redirect `/thank-you` if successful and the user no longer has any votes
  - Redirects to `/` login page and display and error if unsuccessful.

We'll still need to handle the hashing of emails in a follow-up PR, but I think we can safely merge this for now!

Closes #5

### CODE REVIEW

**Comment by ashermanwmf** [comment]
IMO merge or finish query now that DB is set up better. Thoughts?

File: pages/api/login.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+const escape = require('sql-template-strings')
+
+function redirectTo (res, path) {
+  res.writeHead(307, { Location: path })
+  res.end()
+}
+
+async function findUser (email) {
+  // Normalize email by downcasing it and stripping whitespace
+  email = email.toLowerCase().trim()
```
Comment (ashermanwmf) [inline pages/api/login.js:11]: smart

File: pages/api/login.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+  const userResult = await db.query(escape`SELECT * FROM voters WHERE email = ${email} LIMIT 1`)
+  return userResult[0]
+}
+
+module.exports = async (req, res) => {
+  if (req.method === 'POST') {
+    let email = req.body.email;
+    if (email === undefined) {
```
Comment (ashermanwmf) [inline pages/api/login.js:20]: you could do 

if (!email)

if you want

File: pages/index.js [hunk 3]
```diff
@@ -1,9 +1,54 @@
...
+function Index ({ error }) {
+  let errorMessage = error ? (
+    <div class="bg-red-100 border border-red-400 text-red-700 px-3 py-4 mb-3 rounded relative" role="alert">
+      <span class="block sm:inline">{ error }</span>
     </div>
-    <h1 className="text-black-700 leading-normal text-center">Homewreck</h1>
-    <p className="text-gray-500 text-center">I'm an uncoordinated husk of a site</p>
-  </div>
-)
+  ) : null
```
Comment (ashermanwmf) [inline pages/index.js:12]: exact! nice stuff.

**Review by cachilders — APPROVED** [review]
i could kiss you if it was possible while maintaining six feet of separation

**Review by ashermanwmf — APPROVED** [review]
lgtm

**Pull request merged**